addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java

```diff
@@ -681,167 +770,211 @@ private boolean isMe(String idFromDevice) {
-            logger.debug("Process fan speed update (0..7)");
             if (messageParts.length != 5) {
                 logger.debug("FAN;SPD;ACTUAL has unexpected number of parameters: {}", Arrays.toString(messageParts));
                 return;
             }
+            logger.debug("Process fan speed update for {}: {}", thingID(), messageParts[4]);
             PercentType state = convertSpeedToPercent(messageParts[4]);
             updateChannel(CHANNEL_FAN_SPEED, state);
             fanStateMap.put(CHANNEL_FAN_SPEED, state);
         }
 
         private void updateFanDirection(String[] messageParts) {
-            logger.debug("Process fan direction update (FWD/REV)");
             if (messageParts.length != 4) {
                 logger.debug("FAN;DIR has unexpected number of parameters: {}", Arrays.toString(messageParts));
                 return;
             }
+            logger.debug("Process fan direction update for {}: {}", thingID(), messageParts[3]);
             StringType state = new StringType(messageParts[3]);
             updateChannel(CHANNEL_FAN_DIRECTION, state);
             fanStateMap.put(CHANNEL_FAN_DIRECTION, state);
         }
 
         private void updateFanAuto(String[] messageParts) {
-            logger.debug("Process fan auto update (ON/OFF)");
             if (messageParts.length != 4) {
                 logger.debug("FAN;AUTO has unexpected number of parameters: {}", Arrays.toString(messageParts));
                 return;
             }
+            logger.debug("Process fan auto update for {}: {}", thingID(), messageParts[3]);
             OnOffType state = messageParts[3].toUpperCase().equals("ON") ? OnOffType.ON : OnOffType.OFF;
             updateChannel(CHANNEL_FAN_AUTO, state);
             fanStateMap.put(CHANNEL_FAN_AUTO, state);
         }
 
         private void updateFanWhoosh(String[] messageParts) {
-            logger.debug("Process fan whoosh update (ON/OFF)");
             if (messageParts.length != 5) {
                 logger.debug("FAN;WHOOSH has unexpected number of parameters: {}", Arrays.toString(messageParts));
                 return;
             }
+            logger.debug("Process fan whoosh update for {}: {}", thingID(), messageParts[4]);
             OnOffType state = messageParts[4].toUpperCase().equals("ON") ? OnOffType.ON : OnOffType.OFF;
             updateChannel(CHANNEL_FAN_WHOOSH, state);
             fanStateMap.put(CHANNEL_FAN_WHOOSH, state);
         }
 
+        private void updateFanWintermode(String[] messageParts) {
+            if (messageParts.length != 4) {
+                logger.debug("WINTERMODE;STATE has unexpected number of parameters: {}", Arrays.toString(messageParts));
+                return;
+            }
+            logger.debug("Process fan wintermode update for {}: {}", thingID(), messageParts[3]);
+            OnOffType state = messageParts[3].toUpperCase().equals("ON") ? OnOffType.ON : OnOffType.OFF;
```

Comment:
If you would put the literal left and use equalsIgnoreCase your code would be a little bit more resilient.